zerver/openapi/zulip.yaml

```diff
@@ -16976,6 +16988,38 @@ components:
                   The unique message ID. Messages should always be
                   displayed sorted by ID.
     BasicStream:
+      allOf:
+        - $ref: "#/components/schemas/BasicStreamBase"
+        - additionalProperties: false
+          properties:
+            stream_id: {}
+            name: {}
+            description: {}
+            date_created: {}
+            invite_only: {}
+            rendered_description: {}
+            is_web_public: {}
+            stream_post_policy: {}
+            message_retention_days:
+              nullable: true
+            history_public_to_subscribers: {}
+            first_message_id:
+              nullable: true
+            is_announcement_only: {}
+            can_remove_subscribers_group: {}
+            stream_weekly_traffic:
+              type: integer
+              nullable: true
+              description: |
+                The average number of messages sent to the stream in recent weeks,
+                rounded to the nearest integer.
+
+                If `null`, the stream was recently created and there is
+                insufficient data to estimate the average traffic or it
+                is not required to send the traffic data to clients.
```

Comment:
I find this "is not required" wording a bit confusing. If I understand right, this is about a situation where we're choosing not to send the information, right?

Ah, I guess what you're going for is that these are situations where the client isn't expected to need the information. I guess that describes spectators, and stream deactivation events.

I agree we shouldn't try to spell out exactly what the situations are where the server doesn't include this information. Here's one version:

```
                If `null`, no information is provided on the average traffic.
                This can be because the stream was recently created
                and there is insufficient data to make an estimate,
                or because the server wishes to omit this information
                for this client, this realm, or this endpoint or type of event.
```